### ISSUE TITLE
Improve error messages for intra-doc links when there are backticks

### ISSUE THREAD

**Issue by GuillaumeGomez** [rust-lang/rust#87169]
In #87078, we arrived at the following situation:

```
error: incompatible link kind for `Foo::bar`
  --> $DIR/field-ice.rs:5:6
   |
LL | /// [`Foo::bar()`]
   |      ^^^^^^^^^^^^ help: to link to the field, remove the disambiguator: ``Foo::bar``
```

I want to underline this part in particular:

```
``Foo::bar``
```

Having two backticks is very weird. What I suggest instead is to reduce the "^^^^^" underlining part in order to not include backticks in it so then they can be removed from the error message as well, which would give:

```
error: incompatible link kind for `Foo::bar`
  --> $DIR/field-ice.rs:5:6
   |
LL | /// [`Foo::bar()`]
   |       ^^^^^^^^^^ help: to link to the field, remove the disambiguator: `Foo::bar`
```

I'll do it in the next days if no one does until then. :)

cc @jyn514

**Comment by estebank** [comment]
You can fix this by using a `span_suggestion_verbose` instead of `span_suggestion`. It'll take more vertical space, but it's ok.

**Comment by GuillaumeGomez** [comment]
@estebank It's a bit different here because we create the spans ourselves because they come from inside a doc comment. So I'm sure if your suggestion will work out.

Also, in here it's `src/test/rustdoc-ui`, not `src/test/ui` (but otherwise it's the same). ;)

**Comment by estebank** [comment]
@GuillaumeGomez thanks for the correction!

As long as the span was synthesized correctly (including what side of a char the boundary falls on), then just changing the method will work as expected.

**Comment by GuillaumeGomez** [comment]
I had in mind to simply update the span instead in case the string was starting and ending with backticks.

**Comment by jyn514** [comment]
@GuillaumeGomez look at the code that's already there: https://github.com/rust-lang/rust/blob/c49895d9049f67e07e297ee487836a587f69690e/src/librustdoc/passes/collect_intra_doc_links.rs#L2104-L2108
If you think this shouldn't use `verbose`, remove that instead. But I think it's better to show backticks if the user had them there originally.

**Comment by GuillaumeGomez** [comment]
We can take advantage of the span to exclude the backticks from the underlining, so then we can avoid displaying the backticks twice in the suggestion. As for the how, I guess it's fine as long as we get the output we want.

**Comment by GuillaumeGomez** [comment]
Fixed in #87285.
